### PR TITLE
[testharness.js] Hide implementation details

### PR DESCRIPTION
**Pull request by jugglinmike**
Introduce a "private" version of `Test#add_cleanup` so that test harness
status messages can more accurately reflect end user expectations.
Document both the "public" and "private" implementations with in-line
comments.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/6244)

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/242958469)

# Firefox (nightly) #
Testing web-platform-tests at revision 7efa1a278e71ad12bfbb28400208ec7d19b2b2ed
Ignoring 2 changed files:
No files changed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/242958470)

# Sauce (safari) #
Testing web-platform-tests at revision 7efa1a278e71ad12bfbb28400208ec7d19b2b2ed
Ignoring 2 changed files:
No files changed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/242958477)

# Chrome (unstable) #
Testing web-platform-tests at revision 7efa1a278e71ad12bfbb28400208ec7d19b2b2ed
Ignoring 2 changed files:
No files changed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/242958480)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 7efa1a278e71ad12bfbb28400208ec7d19b2b2ed
Ignoring 2 changed files:
No files changed

File: resources/testharness.js [hunk 1]
```diff
@@ -1471,10 +1472,27 @@ policies and contribution forms [3].
         }), timeout * tests.timeout_multiplier);
     }
 
-    Test.prototype.add_cleanup = function(callback) {
+    /*
+     * Private method for registering cleanup functions. `testharnes.js`
```
Comment (jgraham) [inline resources/testharness.js]: Typo `testharnes.js`

**Review by jgraham — APPROVED** [review]
Approved, but please fix the typo and then merge.

**Pull request merged**